src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -19,24 +20,30 @@ namespace Microsoft.Oryx.BuildScriptGenerator
     internal class DefaultBuildScriptGenerator : IBuildScriptGenerator
     {
         private readonly IEnumerable<IProgrammingPlatform> _programmingPlatforms;
-        private readonly ILogger<DefaultBuildScriptGenerator> _logger;
         private readonly IEnvironmentSettingsProvider _environmentSettingsProvider;
+        private readonly IEnumerable<IChecker> _checkers;
+        private readonly ILogger<DefaultBuildScriptGenerator> _logger;
 
         public DefaultBuildScriptGenerator(
             IEnumerable<IProgrammingPlatform> programmingPlatforms,
             IEnvironmentSettingsProvider environmentSettingsProvider,
+            IEnumerable<IChecker> checkers,
             ILogger<DefaultBuildScriptGenerator> logger)
         {
             _programmingPlatforms = programmingPlatforms;
             _environmentSettingsProvider = environmentSettingsProvider;
             _logger = logger;
+            _checkers = checkers;
         }
 
-        public bool TryGenerateBashScript(BuildScriptGeneratorContext context, out string script)
+        public bool TryGenerateBashScript(
+            BuildScriptGeneratorContext context,
+            out string script,
+            List<ICheckerMessage> checkerMessageSink = null)
         {
             script = null;
 
-            var toolsToVersion = new Dictionary<string, string>();
+            var toolsToVersion = new Dictionary<string, string>(); // To be populated by GetBuildSnippets
```

Comment:
Just to be safe (I know this is built by _us_ and not depended on end user input), I would make the key comparison ignore the case...there is a overload to this dictionary